### PR TITLE
Small simplication, removes an err check.

### PR DESCRIPTION
**Pull request by domluna**

### CODE REVIEW

File: algorithmia.go [hunk 1]
```diff
@@ -58,13 +58,8 @@ func (c *Client) Query(user, algo string, input interface{}) (*Response, error)
...
 	defer res.Body.Close()
 
-	decoder := json.NewDecoder(res.Body)
-	if err != nil {
-		return nil, err
-	}
-
 	var response *Response
-	err = decoder.Decode(&response)
+	err = json.NewDecoder(res.Body).Decode(&response)
```
Comment (sjkaliski) [inline algorithmia.go]: I'm thinking we just expand this a bit to:

``` go
decoder := json.NewDecoder(res.Body)
err = decoder.Decode(&response)
if err != nil {
  return nil, err
}
```

Just personal preference.

**Comment by sjkaliski** [comment]
One stylistic comment. But good catch!! Thanks.

**Comment by domluna** [comment]
Yeah, that's cool. Your repo so you're the boss!

**Comment by sjkaliski** [comment]
Great! Will merge when fixed.

**Comment by domluna** [comment]
fixed and squashed.

**Pull request merged**